client/in_memory_provider.go

```diff
@@ -33,26 +33,28 @@ type InMemoryProviderOptions struct {
 	// Repository Key of the configuration repository you wish to read
 	// configs from.
 	RepositoryKey RepositoryKey
-	// Interval at which you want to refresh configs from Lekko.
-	// Recommend 15 to 30 seconds.
-	UpdateInterval time.Duration
 }
 
 type inMemoryProviderType string
 
 const (
 	inMemoryProviderTypeBackend inMemoryProviderType = "backend"
+	inMemoryProviderTypeGit     inMemoryProviderType = "git"
+	inMemoryProviderTypeStatic  inMemoryProviderType = "static"
```

Comment:
i think we should have 3 separate modes because currently in the sidecar, we have
- backend
- static
- static + apikey (ghost mode)

and i want to make the 3rd one explicit. I'm not opposed to `git-local` to make it more clear that this reads from git.